e2e_samples/fabric_dataops_sample/README.md

```diff
@@ -1,107 +1,219 @@
-- An Azure Repo.
-
-### Setting up the Infrastructure
+While Fabric is an end-to-end platform, it works best when integrated with other Azure services such as Application Insights, Azure Key Vault, ADLS Gen2, Microsoft Purview and so. This dependency is also because customers have existing investments in these services and want to leverage them with Fabric.
 
-1. Clone the repository:
+### Deployed resources
+
+The sample deploys both Azure and Fabric resources.
+
+The Azure resources are deployed using Terraform. The sample uses the local backend for storing the Terraform state, but it can be easily modified to use remote backends.
+
+Microsoft Fabric resources are deployed using the [Microsoft Fabric terraform provider](https://registry.terraform.io/providers/microsoft/fabric/latest/docs) whenever possible, or using [Microsoft Fabric REST APIs](https://learn.microsoft.com/rest/api/fabric/articles/) for resources that are still not supported by the terraform provider.
+
+Here is a list of resources that are deployed:
+
+- Azure Resources
+  - Azure Data Lake Storage Gen2 (ADLS Gen2)
+  - Azure Key Vault
+  - Azure Log Analytics Workspace
+  - Azure Application Insights
+  - Optional: Microsoft Fabric Capacity (an existing Capacity can be used)
+- Fabric Resources
+  - Microsoft Fabric Workspace
+  - Microsoft Fabric Lakehouse
+  - Azure Data Lake Storage Gen2 shortcut
+  - Microsoft Fabric Environment
+  - Microsoft Fabric Notebooks
+  - Microsoft Fabric Data pipelines
+
+## How to use the sample
+
+### Pre-requisites
+
+- An Entra user that can access Microsoft Fabric (Free license is enough).
+- An Azure subscription with the following:
+  - The `Microsoft.Fabric` [resource provider](https://learn.microsoft.com/azure/azure-resource-manager/management/resource-providers-and-types#register-resource-provider) has been registered on the Azure subscription.
+  - A resource group to which your user should be granted [Contributor](https://learn.microsoft.com/azure/role-based-access-control/built-in-roles/privileged#contributor) and [User Access Administrator](https://learn.microsoft.com/azure/role-based-access-control/built-in-roles/privileged#user-access-administrator) privileged roles.
+  - A [managed identity](https://learn.microsoft.com/entra/identity/managed-identities-azure-resources/overview) OR a [service principal](https://learn.microsoft.com/entra/identity-platform/app-objects-and-service-principals):
+    - If you **cannot** create a service principal on your own, request the creation of a service principal as per your organization's policy.
+  - Request that a Fabric administrator grant the above service principal or managed identity permission to [use Fabric APIs](https://learn.microsoft.com/fabric/admin/service-admin-portal-developer#service-principals-can-use-fabric-apis).
+  - Grant the service principal or managed identity the `Contributor` and `User Access Administrator` privileged roles on the Azure resource group.
```

Comment:
for "User access administrator", we may need to specify which conditions to 
E.g
Allow user to only assign selected roles to selected principals (fewer privileges) 

**Allow user to assign all roles except privileged administrator roles Owner, UAA, RBAC (Recommended) **

Allow user to assign all roles (highly privileged)

Reply:
included